apple/Handlers/RNFlingHandler.m

```diff
@@ -158,17 +158,225 @@ - (RNGestureHandlerEventExtraData *)eventExtraData:(id)_recognizer
 
 #else
 
+@interface RNBetterSwipeGestureRecognizer : NSGestureRecognizer {
+  dispatch_block_t failFlingAction;
+  int maxDuration;
+  int minVelocity;
+  double defaultAlignmentCone;
+  double axialDeviationCosine;
+  double diagonalDeviationCosine;
+}
+
+@property (atomic, assign) RNGestureHandlerDirection direction;
+@property (atomic, assign) int numberOfTouchesRequired;
+
+- (id)initWithGestureHandler:(RNGestureHandler *)gestureHandler;
+
+@end
+
+@implementation RNBetterSwipeGestureRecognizer {
+  __weak RNGestureHandler *_gestureHandler;
+
+  NSPoint lastPosition;
+  double lastTime;
+}
+
+- (id)initWithGestureHandler:(RNGestureHandler *)gestureHandler
+{
+  if ((self = [super initWithTarget:self action:@selector(handleGesture:)])) {
+    _gestureHandler = gestureHandler;
+
+    maxDuration = 1.0;
+    minVelocity = 700;
+
+    defaultAlignmentCone = 30;
+    axialDeviationCosine = [self coneToDeviation:defaultAlignmentCone];
+    diagonalDeviationCosine = [self coneToDeviation:(90 - defaultAlignmentCone)];
+  }
+  return self;
+}
+
+- (void)handleGesture:(NSPanGestureRecognizer *)gestureRecognizer
+{
+  [_gestureHandler handleGesture:self];
+}
+
+- (void)mouseDown:(NSEvent *)event
+{
+  [super mouseDown:event];
+
+  lastPosition = [self locationInView:self.view];
+  lastTime = CACurrentMediaTime();
+
+  self.state = NSGestureRecognizerStateBegan;
+
+  __weak typeof(self) weakSelf = self;
+
+  failFlingAction = dispatch_block_create(0, ^{
+    __strong typeof(self) strongSelf = weakSelf;
+
+    if (strongSelf) {
+      strongSelf.state = NSGestureRecognizerStateFailed;
+    }
+  });
+
+  dispatch_after(
+      dispatch_time(DISPATCH_TIME_NOW, (int64_t)(maxDuration * NSEC_PER_SEC)),
+      dispatch_get_main_queue(),
+      failFlingAction);
+}
+
+- (void)mouseDragged:(NSEvent *)event
+{
+  [super mouseDragged:event];
+
+  self.state = NSGestureRecognizerStatePossible;
```

Comment:
I assume `mouseDragged` is called after `mouseDown`, if so then this is going backwards in the state machine.

Reply:
The only way to make it work without setting this state to `Possible` is to change it to `Possible` in `mouseDown` and don't change it here. At least this is what I got while experimenting with different state combinations.

Changed in [880e830](https://github.com/software-mansion/react-native-gesture-handler/pull/3028/commits/880e8309ba03a5c35960b086d4d7c63f579aded5)

Reply:
Weird, I'd expect the recognizer to already be in the `Possible` state at that point.